File: app/src/main/java/com/bridou_n/beaconscanner/events/RxBus.java
package com.bridou_n.beaconscanner.events;

import rx.Observable;
import rx.subjects.PublishSubject;
import rx.subjects.SerializedSubject;
import rx.subjects.Subject;

/**
 * Created by bridou_n on 05/10/2016.
 */

public class RxBus {
    private final Subject<Object, Object> _bus = new SerializedSubject<>(PublishSubject.create());

    public void send(Object o) {
        _bus.onNext(o);
    }

    public void sendError(Throwable e) { _bus.onError(e); }

    public Observable<Object> toObserverable() {
        return _bus;
    }

    public boolean hasObservers() {
        return _bus.hasObservers();
    }
}

File: app/src/main/java/com/bridou_n/beaconscanner/events/RxBus.kt
package com.bridou_n.beaconscanner.events

import com.jakewharton.rxrelay2.PublishRelay
import com.jakewharton.rxrelay2.Relay

import io.reactivex.BackpressureStrategy
import io.reactivex.Flowable

/**
 * Created by bridou_n on 05/10/2016.
 */

class RxBus {
    private val _bus = PublishRelay.create<Any>().toSerialized()

    fun send(o: Any) =  _bus.accept(o)

    fun asFlowable(): Flowable<Any>  = _bus.toFlowable(BackpressureStrategy.LATEST)

    fun hasObservers() =  _bus.hasObservers()
}